extensions/hibernate-reactive/runtime/src/main/java/io/quarkus/hibernate/reactive/runtime/ReactiveSessionProducer.java

```diff
@@ -34,13 +34,24 @@
 
     public void disposeStageSession(@Disposes Stage.Session reactiveSession) {
         if (reactiveSession != null) {
-            reactiveSession.close();
+            //We're ignoring the returned CompletionStage!
+            //This could certainly done better but it should be effective enough to
+            //eventually close the session: the connection pool will order the operations
+            //as it won't be able to return connections to other consumers when it's saturated,
+            //ensuring close operations are handled.
+            reactiveSession.close().toCompletableFuture().join();
         }
     }
 
     public void disposeMutinySession(@Disposes Mutiny.Session reactiveSession) {
         if (reactiveSession != null) {
-            reactiveSession.close();
+            //We're ignoring the returned CompletionStage!
```

Comment:
I agree with the change but I think the comment should be:

```suggestion
            //We must not ignore the returned CompletionStage!
```